skbio/metadata/_testing.py

```diff
@@ -904,3 +905,245 @@ def test_has_positional_metadata(self):
         obj = self._positional_metadata_constructor_(
             2, positional_metadata={'foo': [1, 2], 'bar': ['abc', 'def']})
         self.assertTrue(obj.has_positional_metadata())
+
+
+class IntervalMetadataMixinTests:
+    def _set_up(self):
+        self.upper_bound = 9
+        self.im = IntervalMetadata(self.upper_bound)
+        self.intvls = [
+            {'locations': [(0, 1), (2, 9)], 'metadata': {'gene': 'sagA'}},
+            # repeat it
+            {'locations': [(0, 1)], 'metadata': {'gene': ['a'],
+                                                 'product': 'foo'}}]
+
+    def test_constructor_invalid(self):
+        with self.assertRaisesRegex(TypeError,
+                                    'You must provide `IntervalMetadata` '
+                                    'object.'):
+            self._interval_metadata_constructor_(0, '')
+
+    def test_constructor_interval_metadata_len_mismatch(self):
+        for i in [0, 1, 3, 100]:
+            with self.assertRaisesRegex(ValueError, '\(9\).*\(%d\)' % i):
+                self._interval_metadata_constructor_(i, self.im)
+
+    def test_constructor_no_interval_metadata(self):
```

Comment:
You have tests for missing interval metadata but no tests for interval metadata that is present. Can you add the following test cases?

- test that a length-1 object can have interval metadata set thru the constructor (e.g. similar to `test_constructor_with_positional_metadata_len_one`). This is the base case (i.e. the smallest object that can have interval metadata)
- test that a length-greater-than-1 object can have interval metadata set thru the constructor (e.g. similar to `test_constructor_with_positional_metadata_len_greater_than_one`)